app/src/main/java/ru/suslovalex/androidacademyapp/viewmodel/MoviesDetailsViewModel.kt

```diff
@@ -0,0 +1,34 @@
+package ru.suslovalex.androidacademyapp.viewmodel
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import ru.suslovalex.androidacademyapp.data.Movie
+import ru.suslovalex.androidacademyapp.domain.MovieChecker
+import ru.suslovalex.androidacademyapp.domain.MovieResponseResult
+
+class MoviesDetailsViewModel(private val checker: MovieChecker): ViewModel() {
+    private val _movie = MutableLiveData<Movie>()
+    val movie: LiveData<Movie>
+    get() = _movie
+
+    private val _state = MutableLiveData<State>()
```

Comment:
@AleksandrAlex Please create own `state` for this `viewModel`. Using another viewModel state is not a very good solution